src/components/DataTypes/oneOf.tsx

```diff
@@ -1,20 +1,49 @@
+import { OperatorKeys } from '.';
+import { FULL_TEXT_SLUG } from '../Filters/SchemaSieve';
+import { JSONSchema } from '../Renderer/types';
 import type { CreateFilter } from './utils';
 
 export const operators = () => ({
 	is: 'is',
 	is_not: 'is not',
 });
 
-export type OperatorSlug = keyof ReturnType<typeof operators>;
+export type OperatorSlugs = OperatorKeys<typeof operators>;
 
-export const createFilter: CreateFilter<OperatorSlug> = (
+export const createFilter: CreateFilter<OperatorSlugs> = (
 	field,
 	operator,
 	value,
+	subSchema,
 ) => {
 	const operatorSlug = operator.slug;
 
-	if (operatorSlug === 'is') {
+	if (
+		operatorSlug === FULL_TEXT_SLUG &&
+		subSchema.oneOf?.every(
+			(o): o is JSONSchema =>
+				typeof o === 'object' &&
+				typeof o.title === 'string' &&
+				typeof o.const === 'string',
+		)
+	) {
+		return {
+			properties: {
+				[field]:
+					typeof value === 'string'
+						? {
+								enum: subSchema
+									.oneOf!.filter((o) =>
+										o.title!.toLowerCase().includes(value.toLowerCase()),
+									)
+									.map((o) => o.const!),
```

Comment:
Add `type RequiredProps<T, K extends keyof T> = T & Required<Pick<T, K>>;` to the common types, and then we should be able to remove all the `!`:
```suggestion
			(o): o is RequiredProps<JSONSchema, 'title' | 'const'> =>
				typeof o === 'object' &&
				typeof o.title === 'string' &&
				typeof o.const === 'string',
		)
	) {
		return {
			properties: {
				[field]:
					typeof value === 'string'
						? {
								enum: subSchema
									.oneOf.filter((o) =>
										o.title.toLowerCase().includes(value.toLowerCase()),
									)
									.map((o) => o.const),
```

Reply:
`RequiredProps` is uresolved. I'm adding it in the common-types.ts but I'll probably add a common.d.ts file in rendition in a secondary PR

Reply:
We don't have a `common.d.t` in rendition right?
This should be in `src/common-types.ts` just like how `Overwrite` is.
How we do `common.d.ts` in leaf projects, where tsconfig is set to peak global types from there, does not work for libraries. Consumers will probably get errors while trying to compile a library that uses helpers types defined in a `common.d.ts`, unless they "skipLibCheck" or happen to have the same helper types in their own `common.d.ts`.
So, let's keep it in `src/common-types.ts`.